packages/flutter/lib/src/rendering/wrap.dart

```diff
@@ -392,14 +391,15 @@ class RenderWrap extends RenderBox with ContainerRenderObjectMixin<RenderBox, Wr
     while (child != null) {
       final double childWidth = child.getMaxIntrinsicWidth(double.infinity);
       final double childHeight = child.getMaxIntrinsicHeight(childWidth);
-      if (runWidth + childWidth > width) {
-        height += runHeight;
-        if (runCount > 0)
-          height += runSpacing;
-        runCount += 1;
-        runWidth = 0.0;
-        runHeight = 0.0;
-        childCount = 0;
+      // There must be at least one child before we move on to the next run.
+      if (childCount > 0) {
+        final double spaceWidth = childCount > 0 ? spacing : 0;
```

Comment:
This code only executes when `childCount > 0` as per if above. That means spaceWidth will always be set to `spacing`, no?

Reply:
yes, i'll refactor this, thank you